### PR TITLE
TRUNK-5345: Refactor ConditionService to use new functionality (Work In Progress)

### PR DESCRIPTION
**Pull request by segunolalive**
## JIRA TICKET NAME:
[TRUNK-5345: Refactor ConditionService to use new functionality](https://issues.openmrs.org/browse/TRUNK-5345)

### SUMMARY:
Refactor emrapi module's ConditionService to use this new functionality under the hood (e.g. either delegating to openmrs-core, or using the current emrapi approach, depending on the openmrs-core version)

### CODE REVIEW

**Comment by dkayiwa** [comment]
@segunolalive sorry for making you double work, i have actually dug deeper and noticed that we need both api-2.2 and web-2.2
So do you mind, instead of doing the rename like i had said, to just add the web-2.2 as another sub module?

**Comment by segunolalive** [comment]
@dkayiwa Done

File: web-2.2/pom.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    <parent>
+        <groupId>org.openmrs.module</groupId>
+        <artifactId>emrapi</artifactId>
+        <version>1.25.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>emrapi-web-2.2</artifactId>
+    <packaging>jar</packaging>
+    <name>EMR API Module API 2.2</name>
```
Comment (dkayiwa) [inline web-2.2/pom.xml]: Can you replace api with web in the name and description?

**Comment by segunolalive** [comment]
Done

File: web-2.2/pom.xml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    <parent>
+        <groupId>org.openmrs.module</groupId>
+        <artifactId>emrapi</artifactId>
+        <version>1.25.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>emrapi-web-2.2</artifactId>
+    <packaging>jar</packaging>
+    <name>EMR API Module WEB 2.2</name>
```
Comment (dkayiwa) [inline web-2.2/pom.xml]: What i did mean was replacing api with web

Comment (segunolalive) [reply web-2.2/pom.xml]: Yes, it was originally  `EMR API Module API 2.2`

Comment (segunolalive) [reply web-2.2/pom.xml]: Yes, it was originally `EMR API Module API 2.2`

File: web-2.2/pom.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+    <parent>
+        <groupId>org.openmrs.module</groupId>
+        <artifactId>emrapi</artifactId>
+        <version>1.25.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>emrapi-web-2.2</artifactId>
+    <packaging>jar</packaging>
+    <name>EMR WEB Module WEB 2.2</name>
```
Comment (dkayiwa) [inline web-2.2/pom.xml]: Oh i see!
Then it becomes EMR WEB Module 2.2

**Comment by segunolalive** [comment]
@dkayiwa Done

File: web-2.2/pom.xml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    <parent>
+        <groupId>org.openmrs.module</groupId>
+        <artifactId>emrapi</artifactId>
+        <version>1.25.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>emrapi-web-2.2</artifactId>
+    <packaging>jar</packaging>
+    <name>EMR WEB Module 2.2</name>
+    <description>API project for EMRAPI</description>
```
Comment (dkayiwa) [inline web-2.2/pom.xml]: description also becomes WEB 2.2 project for EMRAPI

**Comment by segunolalive** [comment]
Dome

**Pull request merged**